grayskull.c

```diff
@@ -135,6 +135,29 @@ struct TLB_16M_REG {
 
 #define TLB_16M_SIZE_SHIFT 24
 
+
+static struct tt_attribute_data gs_attributes[] = {
```

Comment:
Can you just put a pointer into tenstorrent_device_class and move init/cleanup to device-indepdent code? Terminate the array with __ATTR_NULL.